sim/firesim-lib/src/main/scala/bridges/TracerVBridge.scala

```diff
@@ -45,6 +53,20 @@ class TracerVBridge(insnWidths: TracedInstructionWidths, numInsns: Int) extends
   // def generateTriggerAnnotations(): Unit = TriggerSource(io.triggerCredit, io.triggerDebit)
   def generateTriggerAnnotations(): Unit =
     TriggerSource.evenUnderReset(WireDefault(io.triggerCredit), WireDefault(io.triggerDebit))
+
+  // To placate CheckHighForm, uniquify blackbox module names by using the
+  // bridge's instruction count as a string suffix. This ensures that TracerV
+  // blackboxes with different instruction counts will have different defnames,
+  // preventing FIRRTL CheckHighForm failure when using a chipyard "Hetero"
+  // config. While a black box parameter relaxes the check on leaf field
+  // widths, CheckHighForm does not permit parameterizations of the length of a
+  // Vec enclosing those fields (as is the case here), since the Vec is lost in
+  // a lowered verilog module.
+  //
+  // See https://github.com/firesim/firesim/issues/729.
+  val defnameSuffix = s"_${numInsns}Wide_" + insnWidths.toString.replaceAll("[(),]", "_")
```

Comment:
```suggestion
  def defnameSuffix = s"_${numInsns}Wide_" + insnWidths.toString.replaceAll("[(),]", "_")
```
Doesn't really need to be a `val`, not that the extra memory use really matters in prctice.